### PR TITLE
Add sexy mouse API

### PR DESCRIPTION
**Pull request by maidmaid**
Using the mouse API is boring, e.g.:

```php
$client->getMouse()->mouseMove(
    $client->getCrawler()->filter('#my-element')->getElement(0)->getCoordinates()
);
```

This PR creates some helpful methods like:

```php
$client->getMouse()->mouseMoveTo($cssSelector = '#my-element');
```

cf https://github.com/symfony/panther/pull/117#issuecomment-427752018

### CODE REVIEW

File: src/WebDriver/WebDriverMouse.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+use Facebook\WebDriver\Interactions\Internal\WebDriverCoordinates;
+use Facebook\WebDriver\Internal\WebDriverLocatable;
+use Facebook\WebDriver\WebDriverMouse as BaseWebDriverMouse;
+use Symfony\Component\Panther\Client;
+
+/**
+ * @author Dany Maillard <danymaillard93b@gmail.com>
+ */
+class WebDriverMouse implements BaseWebDriverMouse
```
Comment (dunglas) [inline src/WebDriver/WebDriverMouse.php]:
```suggestion
final class WebDriverMouse implements BaseWebDriverMouse
```

File: src/Client.php [hunk 2]
```diff
@@ -431,6 +432,6 @@ public function getMouse()
             throw new \RuntimeException(sprintf('"%s" does not implement "%s".', \get_class($this->webDriver), WebDriverHasInputDevices::class));
         }
 
-        return $this->webDriver->getMouse();
+        return new WebDriverMouse($this->webDriver->getMouse(), $this);
```
Comment (dunglas) [inline src/Client.php]: Can you also add the return type to `getMouse()` please?

File: src/WebDriver/WebDriverMouse.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+{
+    /**
+     * @var BaseWebDriverMouse
```
Comment (dunglas) [inline src/WebDriver/WebDriverMouse.php]: can be removed

File: src/WebDriver/WebDriverMouse.php [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+     */
+    private $mouse;
+
+    /**
+     * @var Client
```
Comment (dunglas) [inline src/WebDriver/WebDriverMouse.php]: can be removed

File: src/WebDriver/WebDriverMouse.php [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+     */
+    private $client;
+
+    public function __construct(BaseWebDriverMouse $mouse, Client $client)
+    {
+        $this->mouse = $mouse;
+        $this->client = $client;
+    }
+
+    public function click(WebDriverCoordinates $where)
```
Comment (dunglas) [inline src/WebDriver/WebDriverMouse.php]: Can you add the return types to all methods?

File: src/WebDriver/WebDriverMouse.php [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    {
+        $this->mouse->click($where);
+
+        return $this;
```
Comment (dunglas) [inline src/WebDriver/WebDriverMouse.php]: IIRC, we don't have a fluent API anywhere?

File: src/WebDriver/WebDriverMouse.php [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    public function mouseDownTo($cssSelector)
+    {
+        return $this->mouseDown($this->toCoordinates($cssSelector));
+    }
+
+    public function mouseMove(WebDriverCoordinates $where, $x_offset = null, $y_offset = null)
```
Comment (dunglas) [inline src/WebDriver/WebDriverMouse.php]: Should use camel case in Symfony

File: src/WebDriver/WebDriverMouse.php (hunk 6 above)
Comment (maidmaid) [reply src/WebDriver/WebDriverMouse.php]: It's for respecting the the base contract:
```php
    /**
     * @param WebDriverCoordinates $where
     * @return WebDriverMouse
     */
    public function click(WebDriverCoordinates $where);
```
cf https://github.com/facebook/php-webdriver/blob/community/lib/WebDriverMouse.php

File: src/WebDriver/WebDriverMouse.php (hunk 1 above)
Comment (maidmaid) [reply src/WebDriver/WebDriverMouse.php]: Done

File: src/Client.php (hunk 2 above)
Comment (maidmaid) [reply src/Client.php]: Done.

File: src/WebDriver/WebDriverMouse.php (hunk 3 above)
Comment (maidmaid) [reply src/WebDriver/WebDriverMouse.php]: Removed.

File: src/WebDriver/WebDriverMouse.php (hunk 4 above)
Comment (maidmaid) [reply src/WebDriver/WebDriverMouse.php]: Removed.

File: src/WebDriver/WebDriverMouse.php (hunk 5 above)
Comment (maidmaid) [reply src/WebDriver/WebDriverMouse.php]: Done.

File: src/WebDriver/WebDriverMouse.php (hunk 7 above)
Comment (maidmaid) [reply src/WebDriver/WebDriverMouse.php]: Done.

**Comment by maidmaid** [comment]
Tests added :)

**Comment by dunglas** [comment]
Thanks @maidmaid

**Comment by dunglas** [comment]
This API isn't documented yet. As it's a very nice feature, it would be great if you can add at least some lines in the readme @maidmaid!

**Pull request merged**